rasa_sdk/executor.py

```diff
@@ -162,6 +166,8 @@ def __init__(self) -> None:
         self.actions: Dict[Text, Callable] = {}
         self._modules: Dict[Text, TimestampModule] = {}
         self._loaded: Set[Type[Action]] = set()
+        self.domain: Dict[Text, Any] = {}
```

Comment:
Have you considered to use `None` instead of empty dict and empty string below?

Reply:
I thought it was more safe to have these variables be of the type it is expected. It's also how variables above are defined. You want to differentiate a case where if it's `None` it means that there is no Domain information, and if it is `{}` and `""` the Domain information is there, it's just empty Domain, right?

Reply:
Yea, I was thinking about clearly differentiating case when there is a domain (whether it may be empty or not) and when there is no domain info at all.

Reply:
Noted, makes sense to me, I'll update and test it.

Reply:
I had to update the typing to use `Optional[...]` so that `None` value can be accepted. This resulted in additional changes so that the code can pass the Code Quality check, like removing `ActionCall` as typing and using `Dict[Text, Any]` (which is the correct type for `action_call`.